src/js/helpers/cookies.js

```diff
@@ -53,7 +64,14 @@ module.exports = {
       var c = ca[i];
       while (c.charAt(0) === ' ') { c = c.substring(1, c.length); }
       if (c.indexOf(nameEQ) === 0) {
-        return this.decodeData(c.substring(nameEQ.length, c.length));
+        var cookie_content = c.substring(nameEQ.length, c.length);
+        // Decode the content if it is only base64-valid characters
+        // Non-base64 characters indicate that the cookie is in the old format
+        if (/^[A-Za-z0-9+/]+$/.test(cookie_content)) {
+            // Base64 decoding after adding equal sign padding if necessary
+            cookie_content = atob(cookie_content.padEnd(cookie_content.length + (4 - cookie_content.length % 4) % 4, '='));
```

Comment:
We should add a try-catch block around `atob`, just in case.

Reference: https://developer.mozilla.org/en-US/docs/Web/API/atob#exceptions

Reply:
Also, is this formula correct?

```
cookie_content.padEnd(cookie_content.length + (4 - cookie_content.length % 4) % 4, '=')
```

From the formula, it might end up having 3 equal signs at the back `===`. From https://en.wikipedia.org/wiki/Base64, there should only be 0, 1 or 2 equal signs?

Reply:
Added the try-catch block in https://github.com/woocommerce/sourcebuster-js/pull/6/commits/c938536f5e1e374ce3fd512876f5160bd9eb34b2

Reply:
> Also, is this formula correct?

While the formula _could_ technically add 3 `=` signs, all `btoa()` results have lengths that are multiples of 4, with 0, 1, or 2 `=` signs at the end. The formula ensures the final length is a multiple of 4 by checking how many characters are needed to achieve this, which results in adding only 0, 1 or 2 `=` signs.

Reply:
> While the formula _could_ technically add 3 `=` signs, all `btoa()` results have lengths that are multiples of 4, with 0, 1, or 2 `=` signs at the end. 

Yes, I agree with this part. Let's call this "part 1".

> The formula ensures the final length is a multiple of 4 by checking how many characters are needed/missing (which results in _adding only the missing_, original 0, 1 or 2 `=` signs).

Let's call this "part 2". I think I understand that you are seeing part 2 is linked with part 1. 

However, I'm seeing part 2 on its own without part 1. Or, in other words, I'm looking at just the `get` function and thinking how or where it might go wrong. For example, if someone or some codes change the `cookie_content` to become a single letter `"a"`, then our `padEnd` logic here will make the `cookie_content` become `"a==="`, which may cause issues. 

Anyway, I do think this is an unlikely case. I just want to voice it out (because it feels kinda weird looking at just part 2, without looking at part 1). We can choose to be more defensive and modify the formula, or we can leave it as is.

Reply:
> Also, is this formula correct?

@layoutd , I just gave some thoughts about the formula, and I think of the following, what do you think?

```js
const padMe = (string) => {
  return string.padEnd(Math.ceil(string.length / 4) * 4, "=");
};
```

Reply:
Yeah, that's 💯  cleaner, not sure what late night  Base64 investigation and trial-and-error resulted in my take on it 🤔. I've updated it in b507320b7dcaf96bfe35f0dc62d070471bbf76c1